src/compiler/checker.ts

```diff
@@ -31079,6 +31079,9 @@ namespace ts {
                     }
                 }
                 forEach(clause.statements, checkSourceElement);
+                if (compilerOptions.noFallthroughCasesInSwitch && clause.fallthroughFlowNode && isReachableFlowNode(clause.fallthroughFlowNode)) {
+                    grammarErrorOnFirstToken(clause, Diagnostics.Fallthrough_case_in_switch);

```

Comment:
Should this be instead errorOnFirstToken or something similar since this is not error found through syntax walk but because of semantic analysis?

Reply:
Yeah, probably.